### PR TITLE
chore: Add git hooks to enforce formatting on code commit

### PR DESCRIPTION
**Pull request by rxliuli**
ref: https://github.com/preactjs/preact-iso/pull/8#discussion_r1482404249

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,12 +47,36 @@
...
+		],
+		"*.md": [
+			"prettier --write"
+		],
+		"*.json": [
+			"prettier --write"
+		]
+	},
+	"simple-git-hooks": {
+		"pre-commit": "pnpm lint-staged"
```
Comment (rschristian) [inline package.json:69]: Probably shouldn't be pnpm in a repo with a yarn lock file.

**Review by rschristian — COMMENTED** [review]
Would also need the `.editorconfig`.

I don't think this is necessary for a repo of this size, however. It'd be better for you to learn how to configure your editor or selectively stage changes in git.

File: package.json (hunk 1 above)
Comment (rxliuli) [reply package.json:69]: My mistake, I didn't notice that. Maybe it's `npx`?

Comment (rxliuli) [reply package.json]: Fixed

File: package.json [hunk 2]
```diff
@@ -26,7 +26,8 @@
 		"README.md"
 	],
 	"scripts": {
-		"test": "node --experimental-vm-modules node_modules/.bin/jest"
+		"test": "node --experimental-vm-modules node_modules/.bin/jest",
+		"postinstall": "npx simple-git-hooks"
```
Comment (rschristian) [inline package.json:30]: Let's just copy over the [`.editorconfig`](https://github.com/preactjs/wmr/blob/3c5672ecd2f958c8eaf372d33c084dc69228ae3f/.editorconfig), I don't think we need precommit hooks here.

Comment (rxliuli) [reply package.json:30]: Ah, sorry, just curious, why use two different types of configuration files like prettier/.editorconfig? Additionally, postInstall is for automatically adding a git hook, which is no different from the husky configuration in wmr/preact-router, and of course, the automatic hook can also be removed.

Comment (rschristian) [reply package.json:30]: > why use two different types of configuration files like prettier/.editorconfig?

Slightly different tools with different uses/responsibilities.

> Additionally, postInstall is for automatically adding a git hook, which is no different from the husky configuration in wmr/preact-router, and of course, the automatic hook can also be removed.

I'm aware. I wouldn't call it useful in either of those repos either.

> Ah, just found out that the [.editorconfig](https://github.com/preactjs/preact-iso/blob/main/.editorconfig) configuration file already exists.

Whoops. This should address the formatting then as-is.

**Comment by rschristian** [comment]
I'm going to close this out, we already have the `.editorconfig` (apparently) which should handle formatting just fine. Prettier configs, git hooks, etc. are unnecessary here.

Thanks for the PR though.

**Pull request closed without merging**